Doctrine/UserManager.php

```diff
@@ -28,30 +28,41 @@ class UserManager extends BaseUserManager
     /**
      * @var string
      */
+    private $className;
+
+    /**
+     * @var string
+     */
     protected $class;
 
     /**
      * @var ObjectRepository
      */
-    protected $repository;
+    private $repository;
 
     /**
      * Constructor.
      *
      * @param PasswordUpdaterInterface $passwordUpdater
      * @param CanonicalFieldsUpdater   $canonicalFieldsUpdater
      * @param ObjectManager            $om
-     * @param string                   $class
+     * @param string                   $className
      */
-    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $class)
+    public function __construct(PasswordUpdaterInterface $passwordUpdater, CanonicalFieldsUpdater $canonicalFieldsUpdater, ObjectManager $om, $className)
     {
         parent::__construct($passwordUpdater, $canonicalFieldsUpdater);
 
         $this->objectManager = $om;
-        $this->repository = $om->getRepository($class);
+        $this->className = $className;
+    }
 
-        $metadata = $om->getClassMetadata($class);
-        $this->class = $metadata->getName();
+    protected function getRepository()
+    {
+        if (null === $this->repository) {
```

Comment:
I would remove the property entirely, making it more compatible with entity manager resetting (when resetting the entity manager, existing repository instances become invalid as they still reference the old one. The reset entity manager will recreate repositories)